### ISSUE TITLE
setRecord(Map<Id, SObject>) fails when ID is null

### ISSUE THREAD

**Issue by TrangOul** [jongpie/NebulaLogger#576]
### Package Edition of Nebula Logger

Unlocked Package

### Package Version of Nebula Logger

v4.11.11

### New Bug Summary

In `LogEntryEventBuilder.getJson(Map<Id, SObject> recordIdToRecord, Boolean isRecordFieldStrippingEnabled)`, if ID is `null`, it is replaced by `''`. This fails, as a variable of `Id` type cannot have such a value. 

Steps to reproduce:
Run this in anonymous Apex console:
```
Logger.fine('a').setRecord(new Map<Id, SObject>{null => new Account()});
```

Stack trace:
```
Error on line 963, column 1: System.StringException: Invalid id: 
Class.LogEntryEventBuilder.getJson: line 963, column 1
Class.LogEntryEventBuilder.setRecord: line 534, column 1
```

Relevant code (`LogEntryEventBuilder.getJson(Map<Id, SObject> recordIdToRecord, Boolean isRecordFieldStrippingEnabled)`:
```
for (Id recordid : recordIdToRecord.keySet()) {
	SObject record = recordIdToRecord.get(recordId);
	if (recordId == null) {
		recordId = ''; //System.StringException: Invalid id: 
	}
	strippedRecordIdToRecord.put(recordId, record);
}
```

Fix:
```
Map<String, SObject> strippedRecordIdToRecord = new Map<String, SObject>(); //changed key type from Id to String
for (Id recordId : recordIdToRecord.keySet()) {
	SObject record = recordIdToRecord.get(recordId);
	final String recordIdStr = recordId == null ? '' : recordId;
	strippedRecordIdToRecord.put(recordIdStr, record);
}
```

**Comment by jongpie** [comment]
@TrangOul thanks for reporting this + thanks for the helpful code snippets! 🥇  Your fix seems great, I'll try to get this released as a bugfix sometime this week or next week.

**Comment by jongpie** [comment]
@TrangOul I'm a little behind schedule, but this should now be fixed in [the latest release, `v4.12.0`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.12.0). Let me know if you run into any other issues or have any other feedback!

**Comment by TrangOul** [comment]
A quick test (with the snippet from the bug report above) confirmed that this bug does not occur anymore. Thank you @jongpie !
